R/tags.R

```diff
@@ -1056,7 +1056,15 @@ extractPreserveChunks <- function(strval) {
     strval <- paste(strval, collapse = "\n")
 
   # matches contains the index of all the start and end markers
-  matches <- gregexpr(pattern, strval, perl = TRUE)[[1]]
+  startmatches <- gregexpr(startmarker, strval, fixed = TRUE)[[1]]
+  endmatches <- gregexpr(endmarker, strval, fixed = TRUE)[[1]]
+  matches <- c(startmatches, endmatches)
```

Comment:
With this matching both start and end directly, if either start or end does not have a match, then one of the `matches` values will be `-1`.

Before this change, BOTH start and end had to be missing to fail the `No markers? Just return.` check.  Now it is `SOME` marker is missing, so just return.

Maybe we should check if both `startmatches` and `endmatches` ? I don't know of the right behavior at first thought

Reply:
Doesn't the early return here catch the case where both are missing?

https://github.com/rstudio/htmltools/pull/171/files#diff-72abdc1c2c5a6d65974d57ef9bd3666cR1070

Reply:
Yes. Were both are missing.

But if start is defined and end is missing, the ordered `matches[[1]]` will return -1 as the end matches was missing.

Reply:
Ahh, right, addressed by  https://github.com/rstudio/htmltools/pull/171/commits/3a5457dd4e4324f80cc75f592ff8aae0a4043015